src/libAtomVM/nifs.c

```diff
@@ -5764,6 +5768,38 @@ static term nif_code_ensure_loaded(Context *ctx, int argc, term argv[])
     return result;
 }
 
+static term nif_code_get_object_code(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+
+    term module_atom = argv[0];
+    VALIDATE_VALUE(module_atom, term_is_atom);
+    AtomString module_name = globalcontext_atomstring_from_term(ctx->global, module_atom);
+    if (IS_NULL_PTR(module_name)) {
+        return ERROR_ATOM;
+    }
+    Module *module = globalcontext_get_module(ctx->global, module_name);
+    if (IS_NULL_PTR(module)) {
+        return ERROR_ATOM;
+    }
+    assert(module->num_filenames >= 1);
+    struct ModuleFilename filename = module->filenames[0];
+    size_t result_size = TUPLE_SIZE(3) + term_binary_heap_size(module->binary_size) + LIST_SIZE(filename.len, 1);
+    if (UNLIKELY(memory_ensure_free(ctx, result_size) != MEMORY_GC_OK)) {
+        RAISE_ERROR(OUT_OF_MEMORY_ATOM);
+    }
+    // Note: this assumes constness of module->binary and could be use-after-free if we allowed changing module bitcode at runtime.
+    term binary = term_from_literal_binary((void *) module->binary, module->binary_size, &ctx->heap, ctx->global);
+    term filename_term = term_from_string(filename.data, filename.len, &ctx->heap);
+    term result = term_alloc_tuple(3, &ctx->heap);
+
+    term_put_tuple_element(result, 0, module_atom);
```

Comment:
Atoms are const but it's dangerous to do GC and use value not saved in a roots. At least this needs a comment, at most it needs to be in roots array.